lib/widgets/action_sheet.dart

```diff
@@ -167,10 +186,45 @@ class QuoteAndReplyButton extends MessageActionSheetMenuItemButton {
     composeBoxController.contentController
       .registerQuoteAndReplyEnd(PerAccountStoreWidget.of(messageListContext), tag,
         message: message,
-        rawContent: fetchedMessage?.content,
+        rawContent: rawContent,
       );
     if (!composeBoxController.contentFocusNode.hasFocus) {
       composeBoxController.contentFocusNode.requestFocus();
     }
   };
 }
+
+class CopyButton extends MessageActionSheetMenuItemButton {
+  CopyButton({
+    super.key,
+    required super.message,
+    required super.messageListContext,
+  });
+
+  @override get icon => Icons.copy;
+
+  @override get label => 'Copy to clipboard';
+
+  @override get onPressed => (BuildContext context) async {
+    // Close the message action sheet. We won't be showing request progress,
+    // but hopefully it won't take long at all, and
+    // fetchRawContentWithFeedback has a TODO for giving feedback if it does.
+    Navigator.of(context).pop();
+
+    final rawContent = await fetchRawContentWithFeedback(
+      context: messageListContext,
+      messageId: message.id,
+      errorDialogTitle: 'Copying failed',
+    );
+
+    if (rawContent == null) {
+      return;
+    }
+
+    if (!messageListContext.mounted) return;
```

Comment:
nit: the non-parallelism looks a bit odd:
```suggestion
    if (rawContent == null) return;

    if (!messageListContext.mounted) return;
```